File: logic-to-visual/prototype/tools/test_clicks.py
"""
Headless browser test for constraint_flow.html click behavior.
Usage: python3 docs/test_clicks.py
"""

import asyncio

from playwright.async_api import async_playwright


async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()

        errors = []
        page.on("pageerror", lambda e: errors.append(f"PAGE ERROR: {e}"))
        page.on(
            "console",
            lambda m: (
                print(f"  [browser {m.type}] {m.text}") if m.type == "error" else None
            ),
        )

        await page.goto("http://localhost:8765/constraint_flow.html")
        await page.wait_for_timeout(1500)

        if errors:
            print("JS errors detected:", errors)
        else:
            print("No JS errors on load ✓")

        # Test 1: Step button click
        await page.click('button.step-btn[data-step="1"]')
        await page.wait_for_timeout(400)
        title = await page.inner_text("#step-title")
        print(f"Step 1 click → sidebar title: {title!r}")

        # Test 2: Count tagged nodes
        node_info = await page.evaluate(
            """() => {
            const nodes = document.querySelectorAll('.d2-node[data-node-id]');
            return {
                count: nodes.length,
                ids: Array.from(nodes).map(n => n.dataset.nodeId)
            };
        }"""
        )
        print(f"Tagged d2-node count: {node_info['count']}")
        print(f"Node IDs found: {node_info['ids']}")

        # Test 3: Click stores node via JS dispatch
        await page.evaluate(
            """() => {
            const el = document.querySelector('.d2-node[data-node-id="stores"]');
            if (el) el.click();
            else console.error('stores node not found!');
        }"""
        )
        await page.wait_for_timeout(300)
        drawer = await page.evaluate(
            """() => ({
            open: document.getElementById('detail-drawer').classList.contains('open'),
            nodeId: document.getElementById('drawer-node-id').textContent,
            bodyLen: document.getElementById('drawer-body').innerHTML.length
        })"""
        )
        print(f"After stores.click(): drawer={drawer}")

        # Test 4: Click stores.notion_db
        await page.evaluate(
            """() => {
            const el = document.querySelector('.d2-node[data-node-id="stores.notion_db"]');
            if (el) el.click();
            else console.error('stores.notion_db not found!');
        }"""
        )
        await page.wait_for_timeout(200)
        drawer2 = await page.evaluate(
            """() => ({
            open: document.getElementById('detail-drawer').classList.contains('open'),
            nodeId: document.getElementById('drawer-node-id').textContent,
        })"""
        )
        print(f"After stores.notion_db.click(): drawer={drawer2}")

        # Test 5: Step 1 after clicking should auto-trigger, does it dim stores?
        lit_stores = await page.evaluate(
            """() => {
            const el = document.querySelector('.d2-node[data-node-id="stores"]');
            return el ? {
                classes: el.className,
                opacity: el.style.opacity,
                cursor: el.style.cursor
            } : null;
        }"""
        )
        print(f"stores node classes/style: {lit_stores}")

        await browser.close()
        print("\nAll tests complete.")


asyncio.run(main())

asyncio.run(main())

File: logic-to-visual/prototype/tools/test_clicks.mjs
import { chromium } from '/Users/hugoevers/VScode-projects/admonish-1/.playwright-mcp/node_modules/playwright-core/index.js';

const browser = await chromium.launch({ headless: true });
const page = await browser.newPage();

page.on('console', m => console.log('[browser]', m.type(), m.text()));
page.on('pageerror', e => console.error('[page error]', e.message));

await page.goto('http://localhost:8765/constraint_flow.html');
await page.waitForTimeout(1500);

// Test 1: step button click updates sidebar
const step1Btn = page.locator('button.step-btn[data-step="1"]').first();
await step1Btn.click();
await page.waitForTimeout(500);
const stepTitle = await page.locator('#step-title').textContent();
console.log('Step 1 click → title:', stepTitle);

// Test 2: how many .d2-node elements are tagged
const nodeInfo = await page.evaluate(() => {
  const nodes = document.querySelectorAll('.d2-node[data-node-id]');
  return {
    count: nodes.length,
    ids: Array.from(nodes).map(n => n.dataset.nodeId),
  };
});
console.log('Tagged nodes:', nodeInfo.count, nodeInfo.ids);

// Test 3: dispatch click on stores node
await page.evaluate(() => {
  const el = document.querySelector('.d2-node[data-node-id="stores"]');
  if (el) {
    el.click();  // uses the click() method vs dispatchEvent
  }
});
await page.waitForTimeout(300);

const drawerState = await page.evaluate(() => ({
  open: document.getElementById('detail-drawer').classList.contains('open'),
  nodeId: document.getElementById('drawer-node-id').textContent,
  bodySnippet: document.getElementById('drawer-body').innerHTML.substring(0, 100),
}));
console.log('Drawer after stores click:', JSON.stringify(drawerState, null, 2));

// Test 4: also try clicking on extraction container
await page.evaluate(() => {
  const el = document.querySelector('.d2-node[data-node-id="extraction"]');
  if (el) el.click();
});
await page.waitForTimeout(200);
const drawerState2 = await page.evaluate(() => ({
  open: document.getElementById('detail-drawer').classList.contains('open'),
  nodeId: document.getElementById('drawer-node-id').textContent,
}));
console.log('Drawer after extraction click:', JSON.stringify(drawerState2));

// Test 5: check if goStep modifies the detail drawer (it shouldn't auto-open it)
await page.evaluate(() => {
  window.goStep(2, null);
});
await page.waitForTimeout(500);
const afterGoStep = await page.evaluate(() => ({
  drawerOpen: document.getElementById('detail-drawer').classList.contains('open'),
  sidebarTitle: document.getElementById('step-title').textContent,
}));
console.log('After goStep(2):', JSON.stringify(afterGoStep));

await browser.close();
console.log('DONE');
